src/tictactoe/render_board.clj

```diff
@@ -0,0 +1,37 @@
+(ns tictactoe.render-board)
+
+(defn- cell-empty? [cell] (= 0 cell))
+
+(defn- render-empty-cell [] " ")
+
+(defn- render-cell [key player-symbol-mapping]
+  (str "  "
+       (if (not (cell-empty? key))
+         (get player-symbol-mapping key key)
+         (render-empty-cell))
```

Comment:
when would you have an empty cell? Aren't they either player marker or spot markers?